source/equipment/tests/equipment_api_tests.cpp

```diff
@@ -231,19 +231,18 @@ BOOST_FIXTURE_TEST_CASE(equipment_reports_should_have_stop_points_codes, Equipme
     const auto filter = "stop_point.has_code_type(CodeType1)";
     equipment::equipment_reports(pb_creator, filter, 1);
 
-    // with "count = 1", only "sa1" with "stop1" will be returned
-    // hence the expected codes are the one from stop point "stop1"
     BOOST_REQUIRE_EQUAL(pb_creator.equipment_reports_size(), 1);
-    const auto pb_codes =
-        pb_creator.get_response().equipment_reports(0).stop_area_equipments(0).stop_area().stop_points(0).codes();
 
-    vector<pair<string, string>> codes;
-    for (const auto& pb_code : pb_codes) {
-        codes.emplace_back(pb_code.type(), pb_code.value());
+    multiset<string> codes;
+    for (const auto& sae : pb_creator.get_response().equipment_reports(0).stop_area_equipments()) {
+        for (const auto& sp : sae.stop_area().stop_points()) {
+            for (const auto& c : sp.codes()) {
+                codes.emplace(c.value());
+            }
+        }
     }
 
-    vector<pair<string, string>> expected_codes = {{"CodeType1", "code1"}, {"CodeType1", "code2"}};
-
+    multiset<string> expected_codes = {"code1", "code2", "code6"};
```

Comment:
So you get only the ones from line A because of the count=1, right?
It may be worth a comment (to explain why we don't get "code8" too).

Just curious: You know why you didn't get "code6" previously?

Reply:
We were previously populating only codes from the 1st stop point of the 1st stop area (aka. `stop1`). This stop point only has the 2 codes `code1` & `code2`. 
In the new version of the test, I'm gathering codes from **all** stop points of all stop areas. Hence the surfacing of `code6` which belong to 'stop3'.
I also wanted to make sure all codes were present and not duplicated (hence the the multiset).

I did add a comment anyway ;)